OpenRA.Mods.Common/Traits/BotModules/BotModuleLogic/BaseBuilderQueueManager.cs

```diff
@@ -232,6 +233,9 @@ bool TickQueue(IBot bot, ProductionQueue queue)
 						SuppressVisualFeedback = true
 					});
 
+					// After the placement of building, we may need to call the BaseExpansionModules to find a place to expansion.
+					// The condition to trigger this is about the buildings we just built (including the one just placed).
+					// At present, we should expand when tech count + production count >= refinery count + tolerate.
```

Comment:
```suggestion
					// After succesfuly placing a building, nudge BaseExpansionModules to expand.
					// We want to avoid expanding too often, so we make a judgement by counting buildings.
```

Reply:
fixed, and 

> failRetryTicks won't become negative?

this place does not have `failRetryTicks`